monkestation/code/modules/cassettes/cassette_approval.dm

```diff
@@ -131,8 +133,78 @@ GLOBAL_LIST_INIT(cassette_reviews, list())
 	message_admins("[submitter]'s tape has been approved by [user]")
 	logger.Log(LOG_CATEGORY_MUSIC, "[submitter]'s tape has been approved by [user]", list("approver" = user.name, "submitter" = submitter.name))
 	action_taken = TRUE
+	verdict = "APPROVED"
 
 /proc/fetch_review(id)
 	return GLOB.cassette_reviews[id]
 
 #undef ADMIN_OPEN_REVIEW
+
+// Handles UI to manage cassettes.
+/client/proc/review_cassettes() //Creates a verb for admins to open up the ui
+	set name = "Review Cassettes"
+	set desc = "Review this rounds cassettes."
+	set category = "Admin.Game"
+	var/datum/review_cassettes/tgui = new(usr)//create the datum
+	tgui.ui_interact(usr)//datum has a tgui component, here we open the window
+
+/datum/review_cassettes
+	var/client/holder //client of whoever is using this datum
+	var/is_funmin = FALSE
+
+/datum/review_cassettes/New(user)//user can either be a client or a mob due to byondcode(tm)
+	if (istype(user, /client))
+		var/client/user_client = user
+		holder = user_client //if its a client, assign it to holder
+	else
+		var/mob/user_mob = user
+		holder = user_mob.client //if its a mob, assign the mob's client to holder
+
+	is_funmin = check_rights(R_FUN)
+/datum/review_cassettes/ui_state(mob/user)
+	return GLOB.admin_state
+
+/datum/review_cassettes/ui_close()// Don't leave orphaned datums laying around. Hopefully this handles timeouts?
+	qdel(src)
+
+/datum/review_cassettes/ui_interact(mob/user, datum/tgui/ui) // Open UI and update as it remains open.
+	. = ..()
+	if(is_funmin)
+		ui = SStgui.try_update_ui(user, src, ui)
+		if(!ui)
+			ui = new(user, src, "CassetteManager")
+			ui.open()
+
+/datum/review_cassettes/ui_data(mob/user)
+	. = ..()
+	var/list/data = list("cassettes" = list()) // Initialize main data structure with an empty "cassettes" list
+
+	if(isnull(GLOB.cassette_reviews) || !GLOB.cassette_reviews.len)
+		return data
+
+	for(var/cassette_id in GLOB.cassette_reviews)
+		var/datum/cassette_review/cassette = GLOB.cassette_reviews[cassette_id]
+		var/submitters_name = cassette.submitter
+		var/obj/item/tape_obj = cassette.submitted_tape
+		var/reviewed = cassette.action_taken
+		var/verdict = cassette.verdict
+
+		// Add this cassette's data under its cassette_id
+		data["cassettes"][cassette_id] = list(
+			"submitter_name" = submitters_name,
+			"tape_name" = tape_obj.name,
+			"reviewed" = reviewed,
+			"verdict" = verdict,
+		)
+	return data
+
+/datum/review_cassettes/ui_act(action, params)
+	. = ..()
+	if(.)
+		return
+	if(isnull(GLOB.cassette_reviews) || !GLOB.cassette_reviews[action])
+		return
```

Comment:
```suggestion
	if(!GLOB.cassette_reviews[action])
		return
```
`GLOB.cassette_reviews` won't be null

Reply:
Yes I assumed the same thing myself but worker bees was giving me null runtimes for this. So I used this to silence it. If there is actually a better way to address that, it would be great.

Reply:
Entirely possible the list might just be empty. How it would be after this point is reached I wouldn't know but would checking its length fit better?